src/main/java/usi/si/seart/config/MainConfig.java

```diff
@@ -34,26 +34,26 @@ public OkHttpClient httpClient() {
                 .build();
     }
 
+    @Bean
+    public Gson gson() {
+        return new Gson();
+    }
+
     @Bean
     public WebMvcConfigurer webConfigurer() {
         return new WebMvcConfigurer() {
             @Override
             public void addCorsMappings(@NotNull final CorsRegistry registry) {
                 registry.addMapping("/**")
                         .allowedMethods("GET")
-                        .allowedOrigins(
-                                "http://localhost:3030",
-                                "http://localhost:7030",
-                                "https://seart-ghs.si.usi.ch"
-                        )
```

Comment:
This was a change I made to develop, not related to your code. All three should be allowed origins.